packages/forklift-console-plugin/src/modules/Providers/views/migrate/ProvidersCreateVmMigrationPage.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { FC, useContext, useEffect } from 'react';
+import { useHistory } from 'react-router';
+import { useForkliftTranslation } from 'src/utils/i18n';
+import { useImmerReducer } from 'use-immer';
+
+import { ProviderModelGroupVersionKind, ProviderModelRef, V1beta1Provider } from '@kubev2v/types';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { Button, Divider, Flex, FlexItem, PageSection, Title } from '@patternfly/react-core';
+
+import { useToggle } from '../../hooks';
+import { getResourceUrl } from '../../utils';
+
+import { setAvailableProviders } from './actions';
+import { PlansCreateForm } from './PlansCreateForm';
+import { CreateVmMigrationContext } from './ProvidersCreateVmMigrationContext';
+import { createInitialState, reducer } from './reducer';
+
+const ProvidersCreateVmMigrationPage: FC<{
+  namespace: string;
+}> = ({ namespace }) => {
+  const { t } = useForkliftTranslation();
+  const history = useHistory();
+
+  const { data: { selectedIds: selectedVmIds = [], provider: sourceProvider = undefined } = {} } =
+    useContext(CreateVmMigrationContext);
+  // the page was entered directly without choosing the VMs
+  const emptyContext = !selectedVmIds?.length || !sourceProvider;
+  // redirect to provider list if the page is entered directly
+  useEffect(() => {
+    if (emptyContext) {
+      history.push(
+        getResourceUrl({
+          reference: ProviderModelRef,
+          namespace: namespace,
+        }),
+      );
+    }
+  }, [emptyContext]);
+
+  const [state, dispatch] = useImmerReducer(
+    reducer,
+    { namespace, sourceProvider, selectedVmIds },
+    createInitialState,
+  );
+
+  const [providers] = useK8sWatchResource<V1beta1Provider[]>({
+    groupVersionKind: ProviderModelGroupVersionKind,
+    namespaced: true,
+    isList: true,
+    namespace,
+  });
+  useEffect(() => dispatch(setAvailableProviders(providers ?? [])), [providers]);
+
+  const [isLoading, toggleIsLoading] = useToggle();
+  const onUpdate = toggleIsLoading;
+
+  if (emptyContext) {
+    // wait for redirect triggered from useEffect
+    return <></>;
```

Comment:
use some empty page ?

Reply:
This would cause blinking - the `useEffect` hook should trigger right after first render.

Reply:
hmm, can you be more verbose in the comment, make it clear this is the expected behavior because ...